core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java

```diff
@@ -108,8 +115,10 @@ private ByteBufferManager createByteBufferManager() throws InitialisationExcepti
                                                                    BUFFER_MANAGER_FACTORY_CLASS)),
                                         e, this);
     }
-
-    return factory.create();
+    ByteBufferManager byteBufferManager = factory.create();
+    byteBufferManager.setByteBufferProvider(memoryManagementService.getByteBufferProvider(getByteBufferProviderName(), HEAP));
```

Comment:
why keep having the ByteBufferManager at all?  Looks like what the ByteBuferProvider should be able to do everything it does